tower_cli/conf.py

```diff
@@ -399,9 +401,10 @@ def with_global_options(method):
         '-p', '--tower-password',
         help='Password to use to authenticate to Ansible Tower. '
              'This will take precedence over a password provided to '
-             '`tower config`, if any.',
+             '`tower config`, if any. If value is ASK you will be '
+             'prompted for the password',
         required=False, callback=_apply_runtime_setting,
-        is_eager=True
+        is_eager=False
```

Comment:
Sorry I missed this before, but I think that `is_eager=False` won't work.

the reason is because the related lookups are also processed as field callbacks. So that means that when you do something like this:

```
tower-cli project get --organization=Default --tower-password=ASK
```

Then it looks up the related organization using the runtime settings. In the syntax I gave there, click processes the `--organization` option before the `--tower-password` option, which means that the organization will be looked up with the file settings, before the runtime settings have been applied. click supports only a single-tier of custom processing ordering, which is that eager params are processed first.